python/ppml/src/bigdl/ppml/fl/utils.py

```diff
@@ -44,10 +44,9 @@ def update_available_port(self):
 
     def get_available_port(self, port_start, port_end):
         def is_available(p):            
-            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-            result = sock.connect_ex(('127.0.0.1', p))            
-            sock.close()
-            return result != 0
+            with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+                result = sock.connect_ex(('127.0.0.1', p))            
+                return result != 0
```

Comment:
should this return be put outside the with?

Reply:
I'm not sure, please confirm @glorysdj

Reply:
confirmed, it's ok